### ISSUE TITLE
Analyzer doesn't flag using a versioned API from an unversioned assembly

### ISSUE THREAD

**Issue by terrajobst** [dotnet/roslyn-analyzers#4009]
### Repro

```C#
[assembly: SupportedOSPlatform("windows")]

static class Program
{
    public static void Main()
    {
        Some.WindowsSpecificApi();
    }
}

[SupportedOSPlatform("windows10.0")]
static class Some
{
    public static void WindowsSpecificApi()
    {
    }
}
```

### Expected

> 'WindowsSpecificApi' requires windows 10.0 version or later

### Actual

No diagnostic is produced.

**Comment by buyaa-n** [comment]
Fixed, [the test added](https://github.com/dotnet/roslyn-analyzers/pull/3917/commits/28a7cba35522a24494a41dc66f6b12e52d173512#diff-aa2dc7c8b8aa7394d45b1352f85abfbbR736-R759), thanks!
